examples/subplots_axes_and_figures/secondary_axis.py

```diff
@@ -0,0 +1,176 @@
+"""
+==============
+Secondary Axis
+==============
+
+Sometimes we want as secondary axis on a plot, for instance to convert
+radians to degrees on the same plot.  We can do this by making a child
+axes with only one axis visible via `.Axes.axes.secondary_xaxis` and
+`.Axes.axes.secondary_yaxis`.
+
+If we want to label the top of the axes:
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+import matplotlib.dates as mdates
+from matplotlib.transforms import Transform
+from matplotlib.ticker import (
+    AutoLocator, AutoMinorLocator)
+
+fig, ax = plt.subplots(constrained_layout=True)
+x = np.arange(0, 360, 1)
+y = np.sin(2 * x * np.pi / 180)
+ax.plot(x, y)
+ax.set_xlabel('angle [degrees]')
+ax.set_ylabel('signal')
+ax.set_title('Sine wave')
+secax = ax.secondary_xaxis('top')
+plt.show()
+
+###########################################################################
+# However, its often useful to label the secondary axis with something
+# other than the labels in the main axis. In that case we need to provide
+# both a forward and an inverse conversion function in a tuple
+# to the ``conversion`` kwarg:
+
+fig, ax = plt.subplots(constrained_layout=True)
+x = np.arange(0, 360, 1)
+y = np.sin(2 * x * np.pi / 180)
+ax.plot(x, y)
+ax.set_xlabel('angle [degrees]')
+ax.set_ylabel('signal')
+ax.set_title('Sine wave')
+
+
+def deg2rad(x):
+    return x * np.pi / 180
+
+
+def rad2deg(x):
+    return x * 180 / np.pi
+
+secax = ax.secondary_xaxis('top', conversion=(deg2rad, rad2deg))
```

Comment:
Not sure if that is of relevance, but `conversion=(np.deg2rad, np.rad2deg)` would be much simpler.